analyzers/src/SonarAnalyzer.Common/Rules/ParametersCorrectOrderBase.cs

```diff
@@ -18,141 +18,161 @@
-                {
-                    continue;
-                }
-
-                return true;
-            }
+        var parameterNames = argumentParameterMappings.Values
+            .Select(x => x.Name.ToLowerInvariant())
+            .Distinct()
+            .ToList();
 
-            return false;
+        var argumentIdentifiers = argumentList
+            .Select(x => ConvertToArgumentIdentifier(x, analysisContext.SemanticModel))
+            .ToList();
+        var identifierNames = argumentIdentifiers
+            .Select(x => x.IdentifierName?.ToLowerInvariant())
+            .ToList();
 
-            bool IdentifierWithSameNameAndTypeExists(IParameterSymbol parameter) =>
-                argumentIdentifiers.Any(ia =>
-                    ia.IdentifierName == parameter.Name &&
-                    GetArgumentTypeSymbolInfo(ia.ArgumentSyntax, semanticModel).ConvertedType.DerivesOrImplements(parameter.Type));
+        if (parameterNames.Intersect(identifierNames).Any()
+            && HasIncorrectlyOrderedParameters(argumentIdentifiers, argumentParameterMappings, parameterNames, identifierNames, analysisContext.SemanticModel))
+        {
+            // for VB the symbol does not contain the method syntax reference
+            var secondaryLocations = methodSymbol.DeclaringSyntaxReferences
+                .Select(x => GetMethodDeclarationIdentifierLocation(x.GetSyntax()))
+                .WhereNotNull();
+
+            analysisContext.ReportIssue(Diagnostic.Create(rule, getLocationToReport(), secondaryLocations, methodSymbol.Name));
         }
+    }
 
-        private ArgumentIdentifier ConvertToArgumentIdentifier(TArgumentSyntax argument)
+    private bool HasIncorrectlyOrderedParameters(
+        List<ArgumentIdentifier> argumentIdentifiers,
+        Dictionary<TArgumentSyntax, IParameterSymbol> argumentParameterMappings,
+        List<string> parameterNames,
+        List<string> identifierNames, SemanticModel model)
+    {
+        var mappedParams = new HashSet<string>();
+        var mappedArgs = new HashSet<string>();
+        for (var i = 0; i < argumentIdentifiers.Count; i++)
         {
-            var identifierName = GetArgumentIdentifier(argument)?.Text;
-            var nameColonIdentifier = GetNameColonArgumentIdentifier(argument);
+            var argumentIdentifier = argumentIdentifiers[i];
+            var identifierName = argumentIdentifier.IdentifierName?.ToLowerInvariant();
+            var parameter = argumentParameterMappings[argumentIdentifier.ArgumentSyntax];
+            var parameterName = parameter.Name.ToLowerInvariant();
 
-            if (nameColonIdentifier == null)
+            if (string.IsNullOrEmpty(identifierName) || !parameterNames.Contains(identifierName))
             {
-                return new PositionalArgumentIdentifier(identifierName, argument);
+                continue;
```

Comment:
This set of `if` and `continue` is very obfuscating. Try to remove all the `continue` statements by using nested `if` and `if/else if`.
You'll need to invert most of the conditions and it should get easier to understand